string_array.go

```diff
@@ -43,9 +43,7 @@ func (s *stringArrayValue) Replace(val []string) error {
 
 func (s *stringArrayValue) GetSlice() []string {
 	out := make([]string, len(*s.value))
-	for i, d := range *s.value {
-		out[i] = d
-	}
+	copy(out, *s.value)
```

Comment:
while you're doing this, can you also fix `Replace` above?

```diff
--- a/string_array.go
+++ b/string_array.go
@@ -30,13 +30,7 @@ func (s *stringArrayValue) Append(val string) error {
 
 func (s *stringArrayValue) Replace(val []string) error {
        out := make([]string, len(val))
-       for i, d := range val {
-               var err error
-               out[i] = d
-               if err != nil {
-                       return err
-               }
-       }
+       copy(out, val)
        *s.value = out
        return nil
 }
```

[nogo](https://github.com/bazelbuild/rules_go/blob/master/go/nogo.rst) complained about `err` in `Replace` never being used, and while looking into fixing this, I found this change, which is an even better fix than just removing `err` from it.

Reply:
@ixdy mmh, yeah, will look into it. Thanks.

Reply:
@idxy done